src/oemof/solph/components/_offset_converter.py

```diff
@@ -72,111 +74,147 @@ class OffsetConverter(Node):
+    120.0
+
     """  # noqa: E501
 
     def __init__(
         self,
         inputs,
         outputs,
         label=None,
+        conversion_factors=None,
+        normed_offsets=None,
         coefficients=None,
         custom_attributes=None,
     ):
         if custom_attributes is None:
             custom_attributes = {}
+
         super().__init__(
             inputs=inputs,
             outputs=outputs,
             label=label,
             custom_properties=custom_attributes,
         )
 
-        if coefficients is not None:
-            self.coefficients = dict()
-            if isinstance(coefficients, tuple):
-                # TODO: add the correct version in the message
-                msg = (
-                    "Passing a tuple to the keyword `coefficients` will be"
-                    " deprecated in a later version. Please use a dict to"
-                    " specify the corresponding output flow. The first output"
-                    " flow will be assumed as target by default."
-                )
-                warn(msg, DeprecationWarning)
-                if len(coefficients) != 2:
-                    raise ValueError(
-                        "Two coefficients or coefficient series have to be"
-                        " given."
-                    )
-                self.coefficients.update(
-                    {
-                        [k for k in self.outputs.keys()][0]: tuple(
-                            [sequence(i) for i in coefficients]
-                        )
-                    }
-                )
-            elif isinstance(coefficients, dict):
-                for k, v in coefficients.items():
-                    if len(v) != 2:
-                        raise ValueError(
-                            "Two coefficients or coefficient series have to be"
-                            " given."
-                        )
-                    self.coefficients.update(
-                        {k: (sequence(v[0]), sequence(v[1]))}
-                    )
-            else:
+        # this part is used for the transition phase from the old OffsetConverter
+        # API to the new one. It calcualtes the conversion_factors and normed_offsets
+        # from the coefficients and the outputs information on min and max.
+        if coefficients is not None and conversion_factors is None and normed_offsets is None:
+            input_bus = list(inputs.values())[0].input
+            for flow in outputs.values():
+                max = flow.max[0]
+                min = flow.min[0]
+                eta_at_max = max * coefficients[1] / (max - coefficients[0])
+                eta_at_min = min * coefficients[1] / (min - coefficients[0])
+
+                slope, offset = calculate_slope_and_offset_with_reference_to_output(max, min, eta_at_max, eta_at_min)
+                conversion_factors = {input_bus: slope}
+                normed_offsets = {input_bus: offset}
+                msg = ""
```

Comment:
This is the draft for the compatibility adapter to the old API of the `OffsetConverter`. I think I have to make two more changes (`FutureWarning`/`DepreciationWarning` to be added):

- the coefficients may be iterables, so I would have to calculate the slope and the offset correspondingly
- the max and min may also be iterables, if I see it correctly. What assumptions can be taken here: That the user specifies same length iterables for all of the attributes specified as iterables? May the usage crash without a proper message if he does not?

On top of that, are there more things to be taken into consideration?

Then, there is one last concern: The multi-output `OffsetConverter` made it to the `dev` branch very recently only. Can I consider that feature is not used by anyone yet and simply make it crash, e.g. in the case that users may have specified different `min` or `max` values for every output (which does not make sense, but was possible, see #1067)?

I will also add a test for this section in a different script...